### PR TITLE
Add complex64 and complex128 support for `tf.angle` on GPU

### PR DESCRIPTION
**Pull request by yongtang**
In PR #10643, complex64 and complex128 support have been added for `tf.angle` on CPU. However, because of the compiling errors, the complex support on GPU is not enabled yet.

The issue was that, std::arg is not available on nvidia device for GPU. This fix changes to used atan2 instead, which is available in CUDA.

The relevant test cases have bee enabled.

This fix is related to #10643.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by carlthome** [comment]
This would be super useful for any models relying on spectral transforms! Status?

**Comment by yongtang** [comment]
@carlthome The PR has been rebased to resolve merge conflict.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 53 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 14 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 90 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 105 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by laraibzubair18** [comment]
Kuta dala. Harramta

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 14 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: It has been 29 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 45 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 60 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 80 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 96 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 110 days with no activity and the `awaiting review` label has been applied.

**Comment by girving** [comment]
I'm no longer at Google, and don't review pull requests most of the time.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/core/kernels/cwise_ops.h [hunk 1]
```diff
@@ -29,6 +29,28 @@ limitations under the License.
 namespace Eigen {
 namespace internal {
 
+#if GOOGLE_CUDA
```
Comment (rmlarsen) [inline tensorflow/core/kernels/cwise_ops.h]: This should really be fixed in Eigen instead, e.g. here: 

https://bitbucket.org/eigen/eigen/src/729d33d11c81fd023834a2d61f0f280ba9da48c8/Eigen/src/Core/MathFunctions.h#lines-438

**Pull request merged**